[MM-55004] Add warning log message when app run as root

Paul-vrn opened the pull request:
#### Summary

Add a warning log message when Mattermost app is running with root user

#### Ticket Link

Fixes #24996 
JIRA: https://mattermost.atlassian.net/browse/MM-55004

#### Release Note

```release-note
Add warning log message when app run as root 
```

mattermost-build commented:
Hello @Paul-vrn,

Thanks for your pull request! A Core Committer will review your pull request soon. For code contributions, you can learn more about the review process [here](https://developers.mattermost.com/contribute/getting-started/code-review/).

hanzei commented on server/channels/app/server.go:
```diff
@@ -996,6 +996,11 @@ func (s *Server) Start() error {
 	logListeningPort := fmt.Sprintf("Server is listening on %v", listener.Addr().String())
 	mlog.Info(logListeningPort, mlog.String("address", listener.Addr().String()))
 
+	// Check if the command is called by root user
+	if os.Geteuid() == 0 {
+		mlog.Warn("Running as root is not recommended")
+	}
```
Let's move that code closer to other sanity checks. Next to https://github.com/mattermost/mattermost/blob/37da0531919032722dfc8036b293bd1bdc900a7b/server/channels/app/server.go#L415-L417 would be a good place.

hanzei commented on server/channels/app/server.go:
```diff
@@ -996,6 +996,11 @@ func (s *Server) Start() error {
 
+	// Check if the command is called by root user
+	if os.Geteuid() == 0 {
+		mlog.Warn("Running as root is not recommended")
```
```suggestion
		mlog.Warn("Running Mattermost as root is not recommended. Please use a separate user")
```

hanzei reviewed (changes requested):
Thanks for the contribution @Paul-vrn Let's ensure the warning is placed close to other similar messanges :+1:

streamer45 commented:
@hanzei I wonder if we should add the check at the command level. You can technically still run the binary as root without getting a warning if you run any other subcommand. I'd argue that would equally be a bad idea in most cases so maybe we could just move the check to the entry point `cmd/mattermost/main.go`.

Paul-vrn commented:
> @hanzei I wonder if we should add the check at the command level. You can technically still run the binary as root without getting a warning if you run any other subcommand. I'd argue that would equally be a bad idea in most cases so maybe we could just move the check to the entry point `cmd/mattermost/main.go`.

Yeah, that's what I implemented in the beginning. But I saw most of the log messages being generated in the `app/server.go`, so I changed my mind. 😄
I'll make the change depending on what Hanzei suggests!

hanzei commented:
Good point. Most commands are not safe to run as root. Let's move the check.

hanzei commented on server/channels/app/server.go:
```diff
@@ -411,7 +411,7 @@ func NewServer(options ...Option) (*Server, error) {
 		})
 	}
-
+	
```
```suggestion

```

hanzei replied:
Please remove the unrelated change.

hanzei commented on server/cmd/mattermost/main.go:
```diff
@@ -17,6 +19,12 @@ import (
 
 func main() {
+
+	// Check if the command is called by root user
+	if os.Geteuid() == 0 {
+		mlog.Warn("Running Mattermost as root is not recommended. Please use a separate user")
+	}
```
While the code works correctly, I think it should be placed in `root.go` instead to keep `main.go` clean. You can use the `PersistentPreRun` field of `RootCmd`.

streamer45 commented:
@Paul-vrn Looks like our linter is not happy with some minor formatting. Could you run `gofmt -s -w cmd/mattermost/commands/root.go` and commit?

Paul-vrn commented:
> gofmt -s -w cmd/mattermost/commands/root.go

ah yess didn't see, my bad, next time I'll run this command thx!

streamer45 reviewed (approved):
Looks good! Thank you for the contribution @Paul-vrn! :tada:

hanzei commented:
/update-branch

hanzei commented on server/channels/app/server.go:
```diff
@@ -411,7 +411,6 @@ func NewServer(options ...Option) (*Server, error) {
 			http.Redirect(w, r, r.URL.String(), http.StatusFound)
 		})
 	}
-
 	if _, err = url.ParseRequestURI(*s.platform.Config().ServiceSettings.SiteURL); err != nil {
```
Nit: Would you mind removing the unrelated change?

hanzei commented on server/cmd/mattermost/main.go:
```diff
@@ -7,6 +7,7 @@ import (
 	"os"
 
 	"github.com/mattermost/mattermost/server/v8/cmd/mattermost/commands"
+
```
Same here

mattermost-build commented:
E2E tests not automatically triggered, because the PR is not in a mergeable state. Please update the branch with the base branch and resolve outstanding conflicts.

hanzei commented:
Thank you very much for your contribution and for bearing with me while we figure out the best solution @Paul-vrn !

yasserfaraazkhan reviewed (approved):
🎉 
<img width="600" alt="image" src="https://github.com/mattermost/mattermost/assets/15979783/2252ebf2-3331-471a-b1e9-a7a5cda9c84a">
